### PR TITLE
Feature/sort tracks by date newest first

### PR DESCRIPTION
**Pull request by DominikGuzei**
Changes tracks controller#index to return tracks sorted by date (newest first). It also removes the possibility to request ALL tracks, since this is not needed and a performance vulnerability. It also moves the specifics of the database query into the model where it belongs.

### CODE REVIEW

**Comment by stravid** [comment]
Looks great!

File: app/models/track.rb [hunk 1]
```diff
@@ -8,6 +8,10 @@ class Track < ActiveRecord::Base
   validates :rasterized, :presence => true
   validates :title, :presence => true
 
+  def self.sorted_by_date(offset, limit)
```
Comment (tombu) [inline app/models/track.rb]: I don´t see a necessity to pass offset and limit here. I like it more modular, e.g. `Track.sort_by_date.limit(4).offset(0)` in the controller. The line below would then just be `Track.order("created_at DESC")`.

File: spec/support/rspec_matchers/bad_request_matcher.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+RSpec::Matchers.define :be_bad_request do 
+  match do |response| 
+    response.code.to_s =~ /^4/ 
+  end 
+end
```
Comment (tombu) [inline spec/support/rspec_matchers/bad_request_matcher.rb:5]: Nice one!

**Comment by tombu** [comment]
:+1: :wink:

**Pull request merged**